birdhouse/default.env

```diff
@@ -15,7 +15,21 @@ export DATA_PERSIST_ROOT="/data"
 # TODO: use this variable for other references of the log path (only used in the pavics-jupyter-base's .env file for now)
 export PAVICS_LOG_DIR=/var/log/PAVICS
 
-export DELAYED_EVAL=''
+# Allow different public and internal hostname.
+# Default to PAVICS_FQDN, which must be set in env.local.
+# Must use single-quote for delayed eval.
+export PAVICS_FQDN_PUBLIC='${PAVICS_FQDN}'
+
+# Ensure DELAYED_EVAL is initially empty.
+# Only this base default.env is allowed to reset this variable.  All component
+# default.env should only append to DELAYED_EVAL and not reset it like this.
+DELAYED_EVAL=''
```

Comment:
Is there a way to avoid this, like setting it empty within `pavics-compose` before sourcing anything else, to avoid instructions in the form "*don't do this except here*"?

Reply:
I don't think setting it as initially empty is useful actually. $DELAYED_EVAL is initially unset so explicitly setting it to the empty string doesn't add anything. 

Also, we have a comment at the top of this file that describes how DELAYED_EVAL should be used. I don't think it is necessary to elaborate on it here.

Reply:
Maybe my "reset" wording is confusing you guys, sorry.  I actually want to "initialize before use".  Will remove this "reset" wording.

If you guys noticed, almost all other vars in any `default.env` are setting to some default value, but this `DELAYED_EVAL` var is meant to be appended so it has to be initialized somewhere.  

The main `default.env` is the best place to initialize since it's the first config file being read.  

The initialization can not be in `pavics-compose.sh` since it means other external scripts sourcing the config files will not have the proper initialization.

Reply:
>The main `default.env` is the best place to initialize since it's the first config file being read.

I would agree if it was something meant to be overridden by users, like other configuration settings, but I think `DELAYED_EVAL` is more of a control variable used by `pavics-compose` than a setting. 

Can it be set in `read_default_env` instead, or anywhere that is sourced by everything before including all the other components? 
https://github.com/bird-house/birdhouse-deploy/blob/master/birdhouse/read-configs.include.sh#L57-L63

Reply:
Ah that's a much better suggestion, will put the initialization in `read_default_env`.